src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -963,76 +993,136 @@ private Jedis getActiveSentinel() {
+	private static HostAndPort newHostAndPort(String hostname, int port) {
+		return new HostAndPort(hostname, port);
+	}
+
+	private static HostAndPort newHostAndPort(RedisNode node) {
+		return new HostAndPort(node.getHost(), node.getPort());
+	}
+
+	private GenericObjectPoolConfig<Jedis> nullSafePoolConfig(@Nullable GenericObjectPoolConfig<Jedis> poolConfig) {
+		return poolConfig != null ? poolConfig : new JedisPoolConfig();
 	}
 
 	private int getConnectTimeout() {
 		return Math.toIntExact(clientConfiguration.getConnectTimeout().toMillis());
 	}
 
+	private int getReadTimeout() {
+		return Math.toIntExact(clientConfiguration.getReadTimeout().toMillis());
+	}
+
+	@Nullable
+	private String getRedisUsername() {
+		return RedisConfiguration.getUsernameOrElse(this.configuration, standaloneConfig::getUsername);
+	}
+
+	private RedisPassword getRedisPassword() {
+		return RedisConfiguration.getPasswordOrElse(this.configuration, standaloneConfig::getPassword);
+	}
+
 	private MutableJedisClientConfiguration getMutableConfiguration() {
 
 		Assert.state(clientConfiguration instanceof MutableJedisClientConfiguration,
-				() -> String.format("Client configuration must be instance of MutableJedisClientConfiguration but is %s",
-						ClassUtils.getShortName(clientConfiguration.getClass())));
+				() -> "Client configuration must be instance of MutableJedisClientConfiguration but is %s"
+						.formatted(ClassUtils.getShortName(clientConfiguration.getClass())));
 
 		return (MutableJedisClientConfiguration) clientConfiguration;
 	}
 
 	private void assertInitialized() {
 
-		State current = state.get();
+		State state = this.state.get();
 
-		if (State.STARTED.equals(current)) {
+		if (isStarted(state)) {
 			return;
 		}
 
-		switch (current) {
-			case CREATED, STOPPED -> throw new IllegalStateException(
-					String.format("JedisConnectionFactory has been %s. Use start() to initialize it", current));
-			case DESTROYED -> throw new IllegalStateException(
-					"JedisConnectionFactory was destroyed and cannot be used anymore");
-			default -> throw new IllegalStateException(String.format("JedisConnectionFactory is %s", current));
+		switch (state) {
+			case CREATED, STOPPED ->
+					throwIllegalStateException("JedisConnectionFactory has been %s. Call start() to initialize", state);
+			case DESTROYED ->
+					throwIllegalStateException("JedisConnectionFactory was destroyed and cannot be used anymore");
+			default -> throwIllegalStateException("JedisConnectionFactory is %s", state);
 		}
 	}
 
+	private void throwIllegalStateException(String message, Object... args) {
+		throw new IllegalStateException(message.formatted(args));
+	}
+
 	/**
 	 * Mutable implementation of {@link JedisClientConfiguration}.
 	 *
 	 * @author Mark Paluch
 	 */
+	@SuppressWarnings("rawtypes")
 	static class MutableJedisClientConfiguration implements JedisClientConfiguration {
 
-		private boolean useSsl;
```

Comment:
What's the reason to tear code with good locality apart? All SSL-related fields have been kept together. Now, these are spread across.

Reply:
Organization by type.  All the state variables were bunched together making it difficult to read.

If you want better organization around concern, then perhaps a private, internal `SslConfiguration` class used by `MutableJedisClientConfiguration` makes more sense.

Reply:
Reverted for the time being. This has room for improvement.

Reply:
Now I see where we get our influence:

```java
package redis.clients.jedis;

public final class DefaultJedisClientConfig implements JedisClientConfig {

  private final RedisProtocol redisProtocol;

  private final int connectionTimeoutMillis;
  private final int socketTimeoutMillis;
  private final int blockingSocketTimeoutMillis;

  private volatile Supplier<RedisCredentials> credentialsProvider;
  private final int database;
  private final String clientName;

  private final boolean ssl;
  private final SSLSocketFactory sslSocketFactory;
  private final SSLParameters sslParameters;
  private final HostnameVerifier hostnameVerifier;

  //...
```

This is unfortunate. And, this is only marginally better.